docs/Instrumentations.md

```diff
@@ -208,3 +208,99 @@ Captures information from `send` and `onmessage` events.
 |---|---|---|
 |`http.url`|`string`|Websocket URL.|
 |`response_content_length`|`number`|Payload size in bytes.|
+
+## Instrumentation: Socket.io client
+
+This instrumentation generates spans from messages sent using [socket.io](https://socket.io/) client library. The generated spans conform to [OpenTelemetry specifications on messaging systems](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md). This instrumentation is disabled by default as it requires a bit more advanced setup.
+
+> Enabling websocket instrumentation when using socket.io isn't generally recommended.
```

Comment:
```suggestion
> Enabling websocket instrumentation when using socket.io isn't generally recommended.
```

Can we be more specific about why it shouldn't be done?

Reply:
It's mostly due to the reason this instrumentation was made - websocket is basically just a tcp connection so instrumenting it just shows some data went in, some data came out, so it's kinda useless when a better instrumentation that includes what type of message was sent also exists. But technically there's nothing conflicting if someone does enable both

Reply:
When would one prefer websocket over socket.io? It's just because of the messaging data model? If that's the case, why doesn't the websocket instrumentation follow it? Trying to understand if we can replace this note with a more useful recommendation based on instrumentation needs.

Reply:
Hmm so issue that websocket instrumentation has is that it can't really give a good [span name as wanted by the messaging spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#span-name) - since the data can be in any structure as defined by the user (can be a random string `hello`, can be a string with structure that only the dev knows eg `hello/user=1`, can be stringified json, can be raw bytes or [msgpack](https://msgpack.org/), ...) there isn't a good way to extract a name for what data is being sent

Meanwhile socket.io has a low cardinality event name that both client and server use to determine what to do with the message, so we can generate spans with `event_name send/receive` name.

Socket.io uses websocket as once of the transport methods internally so it would cause spans from both instrumentations, where the inner (websocket data sent) span is less useful. Tho socket.io can also do http-polling or comet for sending data when websocket is unavailable, so http spans would be generated then, so I can see removing this warning too since the lower level communication method can appear anyway

Reply:
Then

```suggestion
```